### ISSUE TITLE
The ability to default generated JavaScript to readonly

### ISSUE THREAD

**Issue by BobFrankston** [microsoft/TypeScript#41357]
## Search Terms

readonly generated javascript

## Suggestion

The JavaScript files generated by TypeScript should not normally be modified by the user so they should be able to default to readonly

## Use Cases

I often find myself accidentally editing a JS file instead of a TS because that's where a breakpoint landed. I suspect that this happens because the file is in another directory so is not in the map. 

Since JS looks so much like TS it's easy to inadvertently edit the wrong file and lose the changes next time you look at the TS file. Defaulting to readonly would reduce the risks. In the rare cases where you do want to edit the generated file user can simply remove the readonly flag.

## Examples

I won't keep losing changes due to editing JS, not TS

## Checklist

My suggestion meets these guidelines:

* [x] This wouldn't be a breaking change in existing TypeScript/JavaScript code
* [x] This wouldn't change the runtime behavior of existing JavaScript code
* [x] This could be implemented without emitting different JS based on the types of the expressions
* [x] This isn't a runtime feature (e.g. library functionality, non-ECMAScript syntax with JavaScript output, etc.)
* [x] This feature would agree with the rest of [TypeScript's Design Goals](https://github.com/Microsoft/TypeScript/wiki/TypeScript-Design-Goals).

**Comment by RyanCavanaugh** [comment]
Standard behavior for computer programs is to *not* overwrite read-only files, which we follow. So if we emitted JS as readonly then we wouldn't be able to build a second time.

**Comment by BobFrankston** [comment]
𝗜 𝗷𝘂𝘀𝘁 𝗿𝗮𝗻 𝗶𝗻𝘁𝗼 𝘁𝗵𝗶𝘀 𝗶𝘀𝘀𝘂𝗲 𝗮𝗴𝗮𝗶𝗻.

"Intended behavior" reminds me of the statement "The surgery was a success but the patient died".

Once again I found that I had done editing of a JS file only to lose it when I eventually edited the ts file. It is surprising that tsc will overwrite a newer JS file. 

I'm open to creative suggestions. There is a setting to not show JS files in the file list if there is a corresponding TS file. Perhaps, a similar feature could mark the buffer so it is visually distinct (or a special tab) read-only for editing even if the file is writable.

(There is a separate question of why debugging accidentally opens the JS file even when there is a map)
